DriveKitDriverDataUI/DrivingConditions/Components/DrivingConditionsContext/ViewModels/ConditionsRoadContextViewModel.swift

```diff
@@ -0,0 +1,144 @@
+            tempItems.append(.heavyUrbanTraffic(distance: distance))
+        }
+        if let distance = distanceByRoadContext[.city], distance > 0 {
+            tempItems.append(.city(distance: distance))
+        }
+        if let distance = distanceByRoadContext[.suburban], distance > 0 {
+            tempItems.append(.suburban(distance: distance))
+        }
+        if let distance = distanceByRoadContext[.expressways], distance > 0 {
+            tempItems.append(.expressways(distance: distance))
+        }
+        self.contextItems = tempItems
+        self.viewModelDidUpdate?()
+    }
+}
+
+extension ConditionsRoadContextViewModel: DKContextCard {
+    var items: [DriveKitCommonUI.DKContextItem] {
+        return self.contextItems
+    }
+    
+    var title: String {
+        var cityDistance: Double = 0
+        var expressWayDistance: Double = 0
+        var suburbanDistance: Double = 0
+        for item in contextItems {
+            switch item {
+                case .heavyUrbanTraffic(let distance):
+                    cityDistance += distance
+                case .city(let distance):
+                    cityDistance += distance
+                case .suburban(let distance):
+                    suburbanDistance += distance
+                case .expressways(let distance):
+                    expressWayDistance += distance
+            }
+        }
+        if cityDistance >= suburbanDistance && cityDistance >= expressWayDistance {
+            return "dk_driverdata_drivingconditions_main_city".dkDriverDataLocalized()
+        } else if suburbanDistance >= expressWayDistance {
+            return "dk_driverdata_drivingconditions_main_suburban".dkDriverDataLocalized()
+        } else {
+            return "dk_driverdata_drivingconditions_main_expressways".dkDriverDataLocalized()
+        }
+    }
+    
+    func getContextPercent(_ context: some DriveKitCommonUI.DKContextItem) -> Double {
+        guard let contextItem = context as? ConditionsRoadContextItem else {
+            return 0
+        }
+        let totalDistance = self.contextItems.reduce(0) {
+            $0 + $1.distance
+        }
+        return contextItem.distance / totalDistance
+    }
+
+    func contextCard(_ updateCompletionHandler: (() -> Void)?) {
+        self.viewModelDidUpdate = updateCompletionHandler
+    }
+}
+
+enum ConditionsRoadContextItem {
+    case heavyUrbanTraffic(distance: Double)
+    case city(distance: Double)
+    case suburban(distance: Double)
+    case expressways(distance: Double)
+
+    var distance: Double {
+        switch self {
+            case .heavyUrbanTraffic(let distance):
+                return distance
+            case .city(let distance):
+                return distance
+            case .suburban(let distance):
+                return distance
+            case .expressways(let distance):
+                return distance
+        }
+    }
```

Comment:
Je propose cette version un peu plus concise:

```suggestion
var distance: Double {
        switch self {
            case .heavyUrbanTraffic(let distance),
                 .city(let distance),
                 .suburban(let distance),
                 .expressways(let distance):
                return distance
        }
    }
```

Reply:
proposition prise en compte sur https://github.com/DriveQuantPublic/drivekit-ui-ios/pull/237/commits/eec8fc25d9f74b8a1646f53317263355a2279878